app/models/crop.rb

```diff
@@ -52,11 +52,19 @@ class Crop
   has_many :varieties, class_name: 'Crop', inverse_of: :parent
   belongs_to :parent, class_name: 'Crop', inverse_of: :varieties
 
+  has_and_belongs_to_many :companions,
+                          class_name: 'Crop',
+                          inverse_of: :companions
+
```

Comment:
@simonv3 Oh darn you're right. I even [tried it outside of rails in a small test app to verify](https://github.com/RickCarlino/of_companion_crops). You need to add companions to the `companion` array, and then read them in the `suggested_companions` array. That's not very useful. Hmmm... there has got to be a way to make this work :thinking: